community/kernel/src/test/java/org/neo4j/kernel/impl/store/kvstore/KeyValueMergerTest.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.neo4j.kernel.impl.store.kvstore;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Test;
+
+import static java.util.Arrays.asList;
+
+import static org.junit.Assert.assertEquals;
+
+import static org.neo4j.kernel.impl.store.kvstore.KeyValueMergerTest.Pair.pair;
+
+public class KeyValueMergerTest
+{
+    @Test
+    public void shouldProvideUpdatesWhenNoDataProvided() throws Exception
+    {
+        // given
+        KeyValueMerger merger = new KeyValueMerger( provider(), provider( pair( 14, 1 ),
```

Comment:
I have just figured out that there is no test for the case when they are both empty as well. Maybe there is value to add one?